src/common/snippets/src/lowered/linear_ir.cpp

```diff
@@ -168,62 +176,53 @@ void LinearIR::init_emitters(const std::shared_ptr<TargetMachine>& target) {
 
 ExpressionPtr LinearIR::get_expr_by_node(const std::shared_ptr<Node>& n) const {
     auto found = m_node2expression_map.find(n);
-    return found == m_node2expression_map.end() ? nullptr : found->second;
-}
-
-ExpressionPort LinearIR::get_expr_by_output(const TensorDescriptorPtr& td) const {
-    auto found = m_output2expression_map.find(td);
-    if (found == m_output2expression_map.end())
-        OPENVINO_THROW("Failed to find expression by output tensor descriptor");
+    OPENVINO_ASSERT(found != m_node2expression_map.end(), "The node " + n->get_friendly_name() + " hasn't been found in Linear IR");
     return found->second;
 }
 
-const std::set<ExpressionPort>& LinearIR::get_exprs_by_input(const TensorDescriptorPtr& td) const {
-    auto found = m_input2expression_map.find(td);
-    if (found == m_input2expression_map.end())
-        OPENVINO_THROW("Failed to find expression by input tensor descriptor");
-    return found->second;
+void LinearIR::replace_input(const std::vector<TensorDescriptor>& consumers, const TensorPtr& to) {
+    for (const auto& consumer_input : consumers) {
+        replace_input(consumer_input, to);
+    }
 }
 
-void LinearIR::replace_input(const ExpressionPtr& expr, size_t port, const TensorDescriptorPtr& to) {
+void LinearIR::replace_input(const ExpressionPtr& expr, size_t port, const TensorPtr& to) {
```

Comment:
Minor: We probably don't need this method at all. We can call directly `replace_output(expr->output_port(port), to)`, and it makes perfect sense to limit our interface to working with ports (in a similar way for inputs and outputs).